### PR TITLE
Use context logger for printing the kafka-producer-library error logs.

### PR DESCRIPTION
**Pull request by BerniVarga**

### CODE REVIEW

File: plain_http_producer.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func (c *plainHTTPMessageProducer) SendMessage(uuid string, message queueProduce
 	req, err := http.NewRequest("POST", c.config.Addr+"/notify", strings.NewReader(message.Body))
 	if err != nil {
 		logger.error(fmt.Sprintf("Error creating new request: %v", err.Error()))
```
Comment (catalin-ilea) [inline plain_http_producer.go]: Can you please not log errors in the plain_http_producer? Follow the same pattern and create errors and return them. They will be logged in the message_forwarder.

Comment (BerniVarga) [inline plain_http_producer.go]: Done. :)

**Comment by tosan88** [comment]
:+1:

**Comment by catalin-ilea** [comment]
Can you please remove the transactionId validation logic?

**Comment by catalin-ilea** [comment]
Can you please move extractOriginSystem(headers map[string]string) to plain_http_producer.go ?

**Comment by catalin-ilea** [comment]
Can you please add a test for plain_http_producer?

**Comment by catalin-ilea** [comment]
https://github.com/Financial-Times/coco-kafka-bridge/blob/master/healthchecks.go#L12
Can you please configure some timeouts and connection re-use on this client?

**Comment by catalin-ilea** [comment]
https://github.com/Financial-Times/coco-kafka-bridge/blob/master/healthchecks.go#L94
Response body is not read in this case which means connections are no reused

**Comment by catalin-ilea** [comment]
https://github.com/Financial-Times/coco-kafka-bridge/blob/master/healthchecks.go#L133
Response body is not read which means connections are no reused

**Comment by BerniVarga** [comment]
I made the suggested changes. Please review.

File: healthchecks.go [hunk 2]
```diff
@@ -88,7 +94,10 @@ func checkProxyConnection(address string, authorizationKey string, hostHeader st
 		logger.error(fmt.Sprintf("Healthcheck: Error executing kafka-proxy GET request: %v", err.Error()))
 		return nil, err
 	}
-	defer resp.Body.Close()
+	defer func() {
```
Comment (Zapadlo) [inline healthchecks.go]: Good pattern, will use myself +1

Comment (BerniVarga) [inline healthchecks.go]: Taken from Martin :P

**Comment by catalin-ilea** [comment]
lgtm

**Pull request merged**